subsys/shell/modules/env_service.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <zephyr/shell/shell.h>
+
+#define HELP_ENV_GET   "[NAME]"
+#define HELP_ENV_SET   "NAME \"VALUE\""
+#define HELP_ENV_UNSET "NAME"
+
+static int cmd_env_get(const struct shell *sh, size_t argc, char **argv)
+{
+	const char *name;
+	const char *value;
+	extern char **environ;
+
+	switch (argc) {
+	case 1:
+		/* list all environment variables */
+		if (environ != NULL) {
+			for (char **envp = environ; *envp != NULL; ++envp) {
+				shell_print(sh, "%s", *envp);
+			}
+		}
+		break;
+	case 2:
+		/* list a specific environment variable */
+		name = argv[1];
+		value = getenv(name);
+		if (value != NULL) {
+			shell_print(sh, "%s", value);
+		}
+		break;
+	default:
+		return EXIT_FAILURE;
+	}
+
+	return EXIT_SUCCESS;
+}
+
+static int cmd_env_unset(const struct shell *sh, size_t argc, char **argv)
+{
+	int ret;
+	const char *name;
+
+	if (argc != 2) {
+		return EXIT_FAILURE;
+	}
+
+	name = argv[1];
+
+	ret = unsetenv(name);
+	if (ret == -1) {
+		perror("unsetenv");
+		return EXIT_FAILURE;
+	}
+
+	return EXIT_SUCCESS;
+}
+
+static int cmd_env_set(const struct shell *sh, size_t argc, char **argv)
+{
+	int ret;
+	const char *name;
+	const char *value;
+
+	if (argc != 3) {
+		return EXIT_FAILURE;
+	}
+
+	name = argv[1];
+	value = argv[2];
+
+	ret = setenv(name, value, 1);
```

Comment:
There probably needs to be some additional checks to verify that names are "well conditioned".

E.g. `[_A-Z][_A-Z0-9]+`